common/thorhelper/thorsoapcall.cpp

```diff
@@ -2065,32 +2113,49 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                 {
                     checkTimeLimitExceeded(&remainingMS);
                     Url &connUrl = master->proxyUrlArray.empty() ? url : master->proxyUrlArray.item(0);
-                    socket.setown(blacklist->connect(connUrl.port, connUrl.host, master->logctx, (unsigned)master->maxRetries, master->timeoutMS, master->roxieAbortMonitor));
-                    if (stricmp(url.method, "https") == 0)
+                    ep.setown(new SocketEndpoint(connUrl.host.get(), connUrl.port));
+                    if (strieq(url.method, "https"))
+                        proto = PersistentProtocol::ProtoTls;
+                    bool shouldClose = false;
+                    Linked<ISocket> psock = persistentHandler?persistentHandler->getAvailable(ep, &shouldClose, proto):nullptr;
```

Comment:
I'm planning on adding support for the user specifying the client cert to use.  Can we somehow use the client cert info a as part of the look up for an available socket?  The cert info is added as part of the context inside createSecureSocket().. the change will mean there may be more than one.

Reply:
Well roxie is shared by many ECL developers and we have no control over how they manage their client certificates.  It may never happen but could happen.

Reply:
OK, yes we should take certificate info into consideration. I guess one of us can merge first, then whatever change is needed can go with the latter merge...

Reply:
Right, let's wait on this.  I may remove custom client certificates from my first pass at SOAPCALL secrets.